### PR TITLE
CR-423 First version of Rabbit support for Cucumber tests

### PR DESCRIPTION
**Pull request by PeterBochel**
Adding first working version of Rabbit support for Cucumber tests.

See: https://collaborate2.ons.gov.uk/jira/browse/CR-423

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/common/rabbit/RabbitHelper.java [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    // Connect to rabbit
+    try {
+      this.rabbit = factory.newConnection();
+    } catch (IOException | TimeoutException e) {
+      String errorMessage = "Failed to connect to RabbitMQ";
+      log.error(errorMessage, e);
+      throw new CTPException(Fault.SYSTEM_ERROR, e, errorMessage);
+    }
+
+    NativeRabbitEventSender sender = new NativeRabbitEventSender(rabbitDetails, exchange);
```
Comment (philwhiles) [inline src/main/java/uk/gov/ons/ctp/common/rabbit/RabbitHelper.java]: Overloading the rabbit sender constructor to allow the connection created by the helper would be better

File: src/main/java/uk/gov/ons/ctp/common/rabbit/RabbitHelper.java [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+  private static RabbitHelper rabbitInteraction = null;
+
+  private Connection rabbit;
+  private Channel channel;
+  private String exchange;
+
+  private EventPublisher eventPublisher;
+
+  private RabbitHelper(RabbitConnectionDetails rabbitDetails, String exchange) throws CTPException {
+    ConnectionFactory factory = new ConnectionFactory();
```
Comment (philwhiles) [inline src/main/java/uk/gov/ons/ctp/common/rabbit/RabbitHelper.java:48]: Please make sure that when you add the yml and environment var overloading mech, that it is not simply burned into this code - rather make that mech reusable in common-service

**Review by philwhiles — CHANGES_REQUESTED** [review]
Please see comments

File: src/main/java/uk/gov/ons/ctp/common/rabbit/RabbitHelper.java (hunk 1 above)
Comment (PeterBochel) [reply src/main/java/uk/gov/ons/ctp/common/rabbit/RabbitHelper.java]: Good idea. 
Done.

File: src/main/java/uk/gov/ons/ctp/common/rabbit/RabbitHelper.java (hunk 2 above)
Comment (PeterBochel) [reply src/main/java/uk/gov/ons/ctp/common/rabbit/RabbitHelper.java:48]: Code for reading the rabbit config file now lives in YmlConfigReader.java in the common service project. 
The actual connection details will be in the rabbitmq.yml file which lives in cucumber-common.

**Pull request merged**